src/gui/dialogs/DialogStateful.hpp

```diff
@@ -0,0 +1,184 @@
+#pragma once
+
+#include "IDialog.hpp"
+#include <array>
+#include "DialogRadioButton.hpp"
+#include "marlin_client.hpp"
+#include "dialog_commands.hpp"
+
+#pragma pack(push)
+#pragma pack(1)
+
+//#define DLG_FRAME_ENA 1
+#define DLG_FRAME_ENA 0
+//general foe any number of phases
+class IDialogStateful : public IDialog {
+protected:
+    //dialog flags bitshift
+    enum {
+        DLG_SHI_MOD = 4, // mode shift
+        DLG_SHI_CHG = 14 // change flag shift
+    };
+    enum : uint32_t {
+#if DLG_FRAME_ENA == 1
+        DLG_DRA_FR = 0x0800, // draw frame
+#else
+        DLG_DRA_FR = 0x0000, // draw frame
+#endif                                          //DLG_FRAME_ENA == 1
+        DLG_TXT_CH = 0x2000,                    // text changed
+        DLG_PRO_CH = 0x4000,                    // progress changed
+        DLG_PPR_CH = 0x8000,                    // part progress changed
+        DLG_PRX_CH = (DLG_PRO_CH | DLG_PPR_CH), // some progress changed
+        DLG_PHA_CH = (DLG_PRX_CH | DLG_TXT_CH), // phase changed
+        //dialog flags bitmasks
+        DLG_MSK_MOD = 0x0003,    // mode mask
+        DLG_MSK_CHG = DLG_PHA_CH // change flag mask
```

Comment:
I would prefer to use ALL UPPERCASE only for preprocessor macros.

Reply:
this "flags" are used like preprocessor macros
that enum has no name
enum is used only to enclose those "macros" within class scope
also it is just moved old code